tests/conftest.py

```diff
@@ -52,19 +53,29 @@ def pytest_configure(config):
     config.addinivalue_line("markers", "k8s: mark test to require a Kubernetes cluster")
 
     kubeconfig = config.getoption(KUBECONFIG_OPTION)
-    if kubeconfig:
+    k8s_context = config.getoption(KUBECONTEXT_OPTION)
+
+    if kubeconfig and not k8s_context:
         p = subprocess.run(
-            ["kubectl", "--kubeconfig", kubeconfig, "config", "current-context"],
+            [
+                "kubectl",
+                "--kubeconfig",
+                str(pathlib.Path(kubeconfig).expanduser()),
```

Comment:
Let's include an additional `resolve()` after `expanduser()` to get rid of relative path segments, such as `../`:
```suggestion
                str(pathlib.Path(kubeconfig).expanduser().resolve()),
```